waiter/src/waiter/service_description.clj

```diff
@@ -1045,14 +1063,14 @@
        (-> run-as-user str/blank? not)
        (required-keys-present? description)))
 
-(defn- token-sequence->merged-data
+(defn token-sequence->merged-data
   "Computes the merged token-data using the provided token-sequence and token->token-data mapping.
    It removes the metadata keys from the returned result."
   [token->token-data token-sequence]
   (loop [loop-token-data {}
          [token & remaining-tokens] token-sequence]
     (if token
-      (recur (merge loop-token-data (token->token-data token))
+      (recur (merge-parameters loop-token-data (token->token-data token))
```

Comment:
You're just updating the existing loop/recur here, so it's probably better *not* to change this, but...

It makes me sad to see loop/recur in Clojure code when there's a higher-order function that would work great in its place. E.g., in this case, we really should have used `reduce` to merge all of the token data.